### DISCUSSION TITLE
[Proposal] Parameterless SelectMany() Overload for LINQ

### DISCUSSION THREAD

**Discussion by JBrown526** [dotnet/csharplang#8438] [Language Ideas] [answered]
Add an overload to SelectMany() with no parameters to flatten nested collections:

Currently to do this, you must call:
```cs
List<List<T>> nested = [];
nested.SelectMany(i => i);
```

It seems to me that having to add `i => i` is fairly pointless, and not entirely intuitive. Instead there could be an overload that takes no selector and simply flattens the nested collections.

```cs
public static IEnumerable<TResult> SelectMany<TResult>(this IEnumerable<IEnumerable<TResult>> source)
{
    // Implementation, essentially:
    return source.SelectMany(i => i);
}
```

Which would subsequently be used like so:
```cs
List<List<T>> nested = [];
nested.SelectMany();
```

**Comment by 333fred** [comment, marked as answer]
I believe you're looking for dotnet/runtime, not this repository. API suggestions are [here](https://github.com/dotnet/runtime/issues/new?assignees=&labels=api-suggestion&projects=&template=02_api_proposal.yml&title=%5BAPI+Proposal%5D%3A+).

**Reply by JBrown526** [reply]
Ah gotcha, my bad!

**Reply by colejohnson66** [reply]
See [dotnet/runtime#54220](https://github.com/dotnet/runtime/issues/54220).

**Comment by HaloFour** [comment]
You could also define one yourself:

```cs
public static class SelecyManyExtensions {
    public static IEnumerable<T> SelectMany<T>(this IEnumerable<IEnumerable<T>> items) {
        return items.SelectMany(i => i);
    }
}
```

**Reply by JBrown526** [reply]
Yep I can and have! But this was more a (poorly placed) API request as I don't think that flattening nested collections is such an exotic and uncommon piece of behaviour that it should need to be defined by a developer every time they need it or as part of a third party library.

**Reply by jnm2** [reply]
That's an interesting lens, the BCL as "containing everything that's not exotic."
